File: ios/screens/RecoverIdentity/RecoverIdentityViewManager.m
//
//  RecoverIdentityViewManager.m
//  bitportal
//
//  Created by Terence Ge on 2019/10/3.
//  Copyright © 2019 Facebook. All rights reserved.
//
#import "React/RCTBridgeModule.h"
#import "React/RCTViewManager.h"

@interface RCT_EXTERN_REMAP_MODULE(RecoverIdentityView, RecoverIdentityViewManager, RCTViewManager)

+ (BOOL)requiresMainQueueSetup
{
  return YES;
}

RCT_EXTERN_METHOD(submitFailed:(nonnull NSString *)message)
RCT_EXTERN_METHOD(submitSucceeded)

RCT_EXPORT_VIEW_PROPERTY(onSubmit, RCTBubblingEventBlock)
RCT_EXPORT_VIEW_PROPERTY(loading, BOOL)
@end

File: ios/screens/RecoverIdentity/RecoverIdentityViewManager.swift
//
//  RecoverIdentityViewManager.swift
//  bitportal
//
//  Created by Terence Ge on 2019/10/3.
//  Copyright © 2019 Facebook. All rights reserved.
//

import UIKit

@objc(RecoverIdentityViewManager)
class RecoverIdentityViewManager: RCTViewManager {
  var recoverIdentityView: RecoverIdentityView?
  
  @objc override func view() -> UIView! {
    self.recoverIdentityView = RecoverIdentityView.init()
    return self.recoverIdentityView
  }
  
  @objc func submitFailed(_ message: String) {
    DispatchQueue.main.async {
      self.recoverIdentityView?.submitFailed(message: message)
    }
  }
  
  @objc func submitSucceeded() {
    DispatchQueue.main.async {
      self.recoverIdentityView?.submitSucceeded()
    }
  }
}
